### PR TITLE
Add Slice<T> support for azure-spring-data-cosmos

### LINKED ISSUE #20950: [FEATURE REQ] spring-data-cosmos paging without incurring cost of count query
**Is your feature request related to a problem? Please describe.**
I want to page through a large result set using continuation tokens and the azure-spring-data-cosmos library. However, pageable repository methods always issue a count query to populate the total elements in the page response, and when the query is complex and the result set is large, the count query can be very expensive. 

**Describe the solution you'd like**
I would like to be able to use CosmosPageRequest to fetch a page of data with a continuation token without issuing a count query to populate the total elements. Perhaps you could support methods which return Slice<T> and take CosmosPageRequest.

**Describe alternatives you've considered**
Considered extending CosmosTemplate, but too many fields and methods are defined with private access modifiers. 

**Additional context**
One particular example is adding a sort to a query which pages through a collection with 5 million records caused the cost of the count query to jump from 4 RUs to 180k RUs. Since we don't need to know the total number of elements for our use case this extra query cost is unnecessary.

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Description Added
- [x] Expected solution specified

### PR DESCRIPTION
**Pull request by johnmannix**
Add Slice<T> support for queries that do not require page count. Avoids running the count query, which can be expensive.

Resolves #20950

### CODE REVIEW

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java
Comment (kushagraThapar) [inline sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java:753]: One concern I have with this implementation is that sometimes, the continuation token can be null, within the query, despite having more pages. I would like to confirm my thought with @mbhaskar

**Review by kushagraThapar — COMMENTED** [review]
Thanks @johnmannix  for the contribution. Overall looks good to me. 
Need to confirm a hunch with @mbhaskar  on using `continuationToken != null` condition for checking if there are no more pages coming from the backend. 

Also, as discussed, would be great if we can have some more test cases in `PageableMemoRepositoryIT`

**Comment by kushagraThapar** [comment]
@chenrujun @saragluna - would be good to get your feedback on this PR as well. Please take a look.

**Comment by kushagraThapar** [comment]
@johnmannix - the Spring live integration tests passed, will merge this PR once other folks approve it, thanks again for your contribution .

**Comment by chenrujun** [comment]
Hi, @lzc-1997-abel , please help to review this PR. 🙏

**Comment by lzc-1997-abel** [comment]
LGTM

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java
Comment (mbhaskar) [reply sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java:753]: response continuationToken will be null only for the last page.

Comment (kushagraThapar) [reply sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java:753]: Thanks @mbhaskar !

**Review by kushagraThapar — APPROVED** [review]
LGTM

**Pull request merged**
